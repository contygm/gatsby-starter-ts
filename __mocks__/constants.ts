const glossaryDefinitions: Array<GlossaryElements> = [
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Air',
            letter: 'A',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'three']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Atom',
            letter: 'A',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'one']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Cloud',
            letter: 'C',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two', 'one']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Moon',
            letter: 'M',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test'],
            relatedPosts: ['/blog/test1', '/wiki/test1'],
            tags: ['two']
        }
    },
    {
        html: `<h3>Definition</h3><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        frontmatter: {
            title: 'Window',
            letter: 'W',
            date: '3-2-2021',
            type: 'glossary',
            syllables: 'a·ban·don',
            phonetics: 'əˈbandən',
            similarWords: ['/blog/test', '/wiki/test1'],
            relatedPosts: ['/wiki/test', '/blog/test1'],
            tags: ['three']
        }
    }
];

const featuredOrRelatedPosts: Array<IndexElements> = [
    {
        excerpt:
            'Lemon drops pastry danish halvah. Chupa chups fruitcake powder...',
        fields: {
            slug: '/baking-tutorial/'
        },
        frontmatter: {
            title: 'Baking Tutorial',
            date: '3-2-2021',
            description: 'a baking tutorial',
            tags: ['two', 'three'],
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    {
        excerpt:
            'Cheesecake lemon drops Cheesecake lemon drops pastry danish halvah...',
        fields: {
            slug: '/ten-best-things/'
        },
        frontmatter: {
            title: '10 Best Things',
            description: 'a list of the ten best things',
            tags: ['one', 'two', 'three'],
            date: '2-2-2021',
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    {
        excerpt: 'Chupa chups fruitcake powder dessert gingerbread gummies...',
        fields: {
            slug: '/guide/'
        },
        frontmatter: {
            title: 'Guide',
            description: 'a guide',
            tags: ['one', 'two'],
            date: '5-2-2021',
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    }
];

const previous = {
    fields: {
        slug: '/post-before'
    },
    frontmatter: {
        title: 'Before - A Post About Things'
    }
};

const next = {
    fields: {
        slug: '/post-after'
    },
    frontmatter: {
        title: 'After - A Post About Things'
    }
};

const siteData: SiteMetadata = {
    title: `Starter`,
    siteUrl: `https://www.domain.com`,
    author: {
        name: `Writer B. Author`,
        description:
            'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla quam velit, vulputate eu pharetra nec, mattis ac neque. Duis vulputate commodo lectus, ac blandit elit tincidunt id. Sed rhoncus, tortor sed eleifend tristique, tortor mauris molestie elit'
    },
    businessName: 'The Company',
    phone: '999-999-9999',
    email: 'email@email.com',
    description: 'a super duper site!',
    image: '/nothing.png'
};

export {
    featuredOrRelatedPosts,
    previous,
    next,
    siteData,
    glossaryDefinitions
};
