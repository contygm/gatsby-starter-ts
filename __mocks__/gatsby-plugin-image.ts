import React from 'react';
const plugin = jest.requireActual('gatsby-plugin-image');

const mockImage = ({ imgClassName, ...props } : {imgClassName: string }) =>
    React.createElement('img', {
        ...props,
        className: imgClassName
    });

module.exports = {
    ...plugin,
    GatsbyImage: jest.fn().mockImplementation(mockImage),
    StaticImage: jest.fn().mockImplementation(mockImage)
};
