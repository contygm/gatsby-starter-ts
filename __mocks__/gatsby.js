const React = require('react');
const gatsby = jest.requireActual('gatsby');

// NOTE: This mocks the graphql() function, Link component, and StaticQuery component.
// Allow testing components that use Link or GraphQL
module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            partiallyActive,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to
            })
    ),
    StaticQuery: jest.fn(),
    useStaticQuery: jest.fn()
};
