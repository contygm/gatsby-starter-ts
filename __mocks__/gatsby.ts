import React from 'react';
const gatsby = jest.requireActual('gatsby');

// NOTE: This mocks the graphql() function, Link component, and StaticQuery component.
// Allow testing components that use Link or GraphQL
// https://stackoverflow.com/questions/54941990/how-to-strongly-type-jest-mocks
// https://stackoverflow.com/questions/48759035/mock-dependency-in-jest-with-typescript
module.exports = {
    ...gatsby,
    graphql: jest.fn(),
    Link: jest.fn().mockImplementation(
        // these props are invalid for an `a` tag
        ({
            activeClassName,
            activeStyle,
            getProps,
            innerRef,
            partiallyActive,
            ref,
            replace,
            to,
            ...rest
        }) =>
            React.createElement('a', {
                ...rest,
                href: to
            })
    ),
    StaticQuery: jest.fn(),
    useStaticQuery: jest.fn().mockImplementation(() => ({
        site: {
            siteMetadata: {
                title: `Starter`,
                siteUrl: `https://www.domain.com`,
                author: {
                    name: `Writer B. Author`,
                    description:
                        'Lorem ipsum dolor sit amet, consectetur adipiscing elit. Nulla quam velit, vulputate eu pharetra nec, mattis ac neque. Duis vulputate commodo lectus, ac blandit elit tincidunt id. Sed rhoncus, tortor sed eleifend tristique, tortor mauris molestie elit'
                },
                businessName: 'The Company',
                phone: '999-999-9999',
                email: 'email@email.com',
                description: 'a super duper site!',
                image: '/nothing.png'
            }
        }
    }))
};
