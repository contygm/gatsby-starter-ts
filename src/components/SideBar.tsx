import * as React from 'react';
import FaIconLink from './FaIconLink';
import {
    faMugHot,
    faGhost,
    faEnvelope,
    faMagnifyingGlass
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import HorizontalPostTile from './HorizontalPostTile';
import { getImage } from 'gatsby-plugin-image';

export interface SideBarProps {
    featured: Array<IndexElements>;
    related: Array<IndexElements>;
}

const SideBar = ({ featured, related }: SideBarProps) => {
    return (
        <>
            <aside className="menu sidebar">
                <div className="box">
                    <p className="menu-label">Search the blog</p>
                    <p className="menu-list control has-icons-left">
                        <span className="icon">
                            <FontAwesomeIcon icon={faMagnifyingGlass} />
                        </span>
                        <input
                            className="input is-rounded"
                            type="text"
                            placeholder="A Blog Title..."
                        />
                    </p>
                </div>
                {related && related.length > 0 && (
                    <div className="box">
                        <p className="menu-label">Related posts</p>
                        <ul className="menu-list">
                            {related.length > 0 &&
                                related.map((post) => {
                                    const image = getImage(
                                        post.frontmatter.headerImage
                                    );

                                    return (
                                        <li key={post.frontmatter.title}>
                                            <HorizontalPostTile
                                                title={post.frontmatter.title}
                                                excerpt={post.excerpt}
                                                slug={post.fields.slug}
                                                image={image}
                                            />
                                        </li>
                                    );
                                })}
                        </ul>
                    </div>
                )}
                <div className="box">
                    <p className="menu-label">Featured posts</p>
                    <ul className="menu-list">
                        {featured.map((post) => {
                            const image = getImage(
                                post.frontmatter.headerImage
                            );
                            return (
                                <li key={post.frontmatter.title}>
                                    <HorizontalPostTile
                                        title={post.frontmatter.title}
                                        slug={post.fields.slug}
                                        excerpt={post.excerpt}
                                        image={image}
                                    />
                                </li>
                            );
                        })}
                    </ul>
                </div>
            </aside>
            {/* sticky social media */}
            <div className="menu sticky box mt-4">
                <p className="menu-label">Share this post</p>
                <ul className="has-text-black has-text-centered columns">
                    <div className="column">
                        <FaIconLink
                            faIcon={faMugHot}
                            color="black"
                        />
                    </div>
                    <div className="column">
                        <FaIconLink
                            faIcon={faGhost}
                            color="black"
                        />
                    </div>
                    <div className="column">
                        <FaIconLink
                            faIcon={faEnvelope}
                            color="black"
                        />
                    </div>
                </ul>
            </div>
        </>
    );
};

export default SideBar;
