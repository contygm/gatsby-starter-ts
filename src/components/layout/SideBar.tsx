import React from 'react';
import { SideBarPostTile } from './SideBarPostTile';
import { getImage } from 'gatsby-plugin-image';

/**
 * Vertical sidebar for post templates
 * @property {PostType} type - the page title to be displayed
 * @property {IndexElements[]} featured - post(s) to be used in the featured section
 * @property {IndexElements[]} related - post(s) to be used in the related section
 *
 * @see Sidebar
 * @category Components
 */
export interface SideBarProps {
    type: PostType;
    featured: IndexElements[];
    related: IndexElements[];
}

/**
 * A vertical sidebar component with optional featured and related posts. Related posts sit
 * on top of the featured posts.
 * @category Components
 *
 * @param {SideBarProps} - props including featured posts, related posts, and type of post
 *
 * @see SideBarProps
 */
export const SideBar = ({ featured, related, type }: SideBarProps) => {
    // TODO: working search requires queries, tabling for now
    // const searchPlaceholder = type === 'glossary' ? 'A word...' : 'Title...';
    const firstSectionTitle =
        type === 'glossary' ? 'Featured Blog Posts' : `Related ${type}s`;
    const firstSectionSlugPrefix = type === 'glossary' ? '/blog' : `/${type}`;
    const secondSectionTitle =
        type === 'glossary' ? 'Featured Wiki Posts' : `Featured ${type}s`;
    const secondSectionSlugPrefix = type === 'glossary' ? '/wiki' : `/${type}`;

    return (
        <aside className="menu sidebar">
            {/* <div className="box">
                <p className="menu-label">{`Search the ${type}`}</p>
                <p className="menu-list control has-icons-left">
                    <span className="icon">
                        <FontAwesomeIcon icon={faMagnifyingGlass} />
                    </span>
                    <input
                        className="input is-rounded"
                        type="text"
                        placeholder={searchPlaceholder}
                        aria-label="blog-search"
                    />
                </p>
            </div> */}
            {related && related.length > 0 && (
                <div className="box">
                    <p className="menu-label">{firstSectionTitle}</p>
                    <ul className="menu-list">
                        {related.length > 0 &&
                            related.map((post) => {
                                const image = getImage(
                                    post.frontmatter.headerImage
                                );

                                return (
                                    <li key={post.frontmatter.title}>
                                        <SideBarPostTile
                                            title={post.frontmatter.title}
                                            excerpt={post.excerpt}
                                            slug={`${firstSectionSlugPrefix}${post.fields.slug}`}
                                            image={image}
                                        />
                                    </li>
                                );
                            })}
                    </ul>
                </div>
            )}
            <div className="box">
                <p className="menu-label">{secondSectionTitle}</p>
                <ul className="menu-list">
                    {featured.map((post) => {
                        const image = getImage(post.frontmatter.headerImage);
                        return (
                            <li key={post.frontmatter.title}>
                                <SideBarPostTile
                                    title={post.frontmatter.title}
                                    slug={`${secondSectionSlugPrefix}${post.fields.slug}`}
                                    excerpt={post.excerpt}
                                    image={image}
                                />
                            </li>
                        );
                    })}
                </ul>
            </div>
        </aside>
    );
};
