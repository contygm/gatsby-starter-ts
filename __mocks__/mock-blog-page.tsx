/* istanbul ignore file */
import { BlogIndexProps } from '../src/pages/blog';
import { mockHeadProps, mockPageProps } from '../src/utils/test-props';
import { HeadProps, PageProps } from 'gatsby';
import { featuredOrRelatedPosts, siteData } from './constants';

const mockBlogs: BlogIndexProps = {
    site: siteData,
    allTags: {
        group: [
            { fieldValue: 'three', totalCount: 3 },
            { fieldValue: 'two', totalCount: 2 },
            { fieldValue: 'one', totalCount: 1 }
        ]
    },
    index: {
        totalCount: 3,
        nodes: featuredOrRelatedPosts
    },
    featured: {
        nodes: featuredOrRelatedPosts
    }
};

export const mockBlogPageData = mockPageProps(
    mockBlogs
) as PageProps<BlogIndexProps>;

export const mockBlogHeadData = mockHeadProps(
    mockBlogs
) as HeadProps<BlogIndexProps>;
