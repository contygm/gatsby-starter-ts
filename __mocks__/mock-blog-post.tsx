/* istanbul ignore file */
import { HeadProps, PageProps } from 'gatsby';
import { BlogPostProps } from '../src/templates/blog-post';
import { mockHeadProps, mockPageProps } from '../src/utils/test-props';
import { previous, next, featuredOrRelatedPosts, siteData } from './constants';

export const mockBlogPost: BlogPostProps = {
    site: {
        siteMetadata: siteData
    },
    markdownRemark: {
        id: 'slkj-dsjflks-dflkjl',
        excerpt:
            'Lemon drops pastry danish halvah. Chupa chups fruitcake powder...',
        html: `<h2>Header</h2><p>Lemon drops pastry danish halvah. Lemon drops pastry danish halvah. Lemon drops pastry danish halvah. Lemon drops pastry danish halvah.</p>`,
        tableOfContents: `<ul><li><p><a href="#liquorice-topping">Liquorice topping</a></p><ul><li><p><a href="#pastry-halvah-caramels">Pastry halvah caramels</a></p><ul> <li><a href="#wafer-oat-cake">Wafer oat cake</a></li></ul></li></ul></li></ul>`,
        frontmatter: {
            date: 'December 02, 2022',
            title: 'Baking Tutorial',
            description: 'a baking tutorial',
            related: ['/one-url/', '/two-url/', '/three-url/'],
            tags: ['one', 'two', 'three'],
            headerImage: {
                childImageSharp: {
                    gatsbyImageData: {}
                }
            }
        }
    },
    previous: previous,
    next: next,
    featured: {
        nodes: featuredOrRelatedPosts
    },
    related: {
        nodes: featuredOrRelatedPosts
    }
};

export const mockBlogPostData = mockPageProps(
    mockBlogPost
) as PageProps<BlogPostProps>;
export const mockBlogPostHeadData = mockHeadProps(
    mockBlogPost
) as HeadProps<BlogPostProps>;
