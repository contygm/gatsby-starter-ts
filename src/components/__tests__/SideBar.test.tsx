import React from 'react';
import { render } from '@testing-library/react';
import { SideBar, SideBarProps } from '../SideBar';
import { mockBlogPost } from '../../../__mocks__/mock-blog-post';

const mockData: SideBarProps = {
    featured: mockBlogPost.featured.nodes,
    related: mockBlogPost.related.nodes
};

describe('SideBar', () => {
    let testProps: SideBarProps;

    beforeEach(() => {
        testProps = mockData;
    });

    it('renders correctly', () => {
        const { asFragment } = render(<SideBar {...testProps} />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('renders correctly without related posts', () => {
        testProps.related = [];
        const { asFragment } = render(<SideBar {...testProps} />);
        expect(asFragment()).toMatchSnapshot();
    });
});
