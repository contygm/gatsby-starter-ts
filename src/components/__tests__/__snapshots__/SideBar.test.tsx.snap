// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideBar renders correctly 1`] = `
<DocumentFragment>
  <aside
    class="menu sidebar"
  >
    <div
      class="box"
    >
      <p
        class="menu-label"
      >
        Search the blog
      </p>
      <p
        class="menu-list control has-icons-left"
      >
        <span
          class="icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          class="input is-rounded"
          placeholder="A Blog Title..."
          type="text"
        />
      </p>
    </div>
    <div
      class="box"
    >
      <p
        class="menu-label"
      >
        Related posts
      </p>
      <ul
        class="menu-list"
      >
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/baking-tutorial/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    Baking Tutorial
                  </h6>
                  <p>
                    Lemon drops pastry danish halvah. Chupa chups fruitcake powder...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/ten-best-things/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    10 Best Things
                  </h6>
                  <p>
                    Cheesecake lemon drops Cheesecake lemon drops pastry danish halvah...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/guide/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    Guide
                  </h6>
                  <p>
                    Chupa chups fruitcake powder dessert gingerbread gummies...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
    <div
      class="box"
    >
      <p
        class="menu-label"
      >
        Featured posts
      </p>
      <ul
        class="menu-list"
      >
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/baking-tutorial/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    Baking Tutorial
                  </h6>
                  <p>
                    Lemon drops pastry danish halvah. Chupa chups fruitcake powder...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/ten-best-things/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    10 Best Things
                  </h6>
                  <p>
                    Cheesecake lemon drops Cheesecake lemon drops pastry danish halvah...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/guide/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    Guide
                  </h6>
                  <p>
                    Chupa chups fruitcake powder dessert gingerbread gummies...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </aside>
  <div
    class="menu sticky box mt-4"
  >
    <p
      class="menu-label"
    >
      Share this post
    </p>
    <ul
      class="has-text-black has-text-centered columns"
    >
      <div
        class="column"
      >
        <li
          class="faicon-link-container"
        >
          <a
            aria-label="social media link"
            class="faicon-link no-label is-black"
            data-cy="faicon-link"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-mug-hot fa-xl icon"
              data-icon="mug-hot"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </div>
      <div
        class="column"
      >
        <li
          class="faicon-link-container"
        >
          <a
            aria-label="social media link"
            class="faicon-link no-label is-black"
            data-cy="faicon-link"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-ghost fa-xl icon"
              data-icon="ghost"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </div>
      <div
        class="column"
      >
        <li
          class="faicon-link-container"
        >
          <a
            aria-label="social media link"
            class="faicon-link no-label is-black"
            data-cy="faicon-link"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-xl icon"
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;

exports[`SideBar renders correctly without related posts 1`] = `
<DocumentFragment>
  <aside
    class="menu sidebar"
  >
    <div
      class="box"
    >
      <p
        class="menu-label"
      >
        Search the blog
      </p>
      <p
        class="menu-list control has-icons-left"
      >
        <span
          class="icon"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-magnifying-glass "
            data-icon="magnifying-glass"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M416 208c0 45.9-14.9 88.3-40 122.7L502.6 457.4c12.5 12.5 12.5 32.8 0 45.3s-32.8 12.5-45.3 0L330.7 376c-34.4 25.2-76.8 40-122.7 40C93.1 416 0 322.9 0 208S93.1 0 208 0S416 93.1 416 208zM208 352c79.5 0 144-64.5 144-144s-64.5-144-144-144S64 128.5 64 208s64.5 144 144 144z"
              fill="currentColor"
            />
          </svg>
        </span>
        <input
          class="input is-rounded"
          placeholder="A Blog Title..."
          type="text"
        />
      </p>
    </div>
    <div
      class="box"
    >
      <p
        class="menu-label"
      >
        Featured posts
      </p>
      <ul
        class="menu-list"
      >
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/baking-tutorial/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    Baking Tutorial
                  </h6>
                  <p>
                    Lemon drops pastry danish halvah. Chupa chups fruitcake powder...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/ten-best-things/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    10 Best Things
                  </h6>
                  <p>
                    Cheesecake lemon drops Cheesecake lemon drops pastry danish halvah...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
        <li>
          <a
            class="horizontal-post-tile"
            href="/blog/guide/"
          >
            <div
              class="media"
            >
              <figure
                class="media-left my-auto"
              >
                <img
                  alt="blog image"
                  image="[object Object]"
                />
              </figure>
              <div
                class="media-content"
              >
                <div
                  class="content"
                >
                  <h6>
                    Guide
                  </h6>
                  <p>
                    Chupa chups fruitcake powder dessert gingerbread gummies...
                  </p>
                </div>
              </div>
            </div>
          </a>
        </li>
      </ul>
    </div>
  </aside>
  <div
    class="menu sticky box mt-4"
  >
    <p
      class="menu-label"
    >
      Share this post
    </p>
    <ul
      class="has-text-black has-text-centered columns"
    >
      <div
        class="column"
      >
        <li
          class="faicon-link-container"
        >
          <a
            aria-label="social media link"
            class="faicon-link no-label is-black"
            data-cy="faicon-link"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-mug-hot fa-xl icon"
              data-icon="mug-hot"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M88 0C74.7 0 64 10.7 64 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C120.5 112.3 128 119.9 128 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C119.5 47.7 112 40.1 112 24c0-13.3-10.7-24-24-24zM32 192c-17.7 0-32 14.3-32 32V416c0 53 43 96 96 96H288c53 0 96-43 96-96h16c61.9 0 112-50.1 112-112s-50.1-112-112-112H352 32zm352 64h16c26.5 0 48 21.5 48 48s-21.5 48-48 48H384V256zM224 24c0-13.3-10.7-24-24-24s-24 10.7-24 24c0 38.9 23.4 59.4 39.1 73.1l1.1 1C232.5 112.3 240 119.9 240 136c0 13.3 10.7 24 24 24s24-10.7 24-24c0-38.9-23.4-59.4-39.1-73.1l-1.1-1C231.5 47.7 224 40.1 224 24z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </div>
      <div
        class="column"
      >
        <li
          class="faicon-link-container"
        >
          <a
            aria-label="social media link"
            class="faicon-link no-label is-black"
            data-cy="faicon-link"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-ghost fa-xl icon"
              data-icon="ghost"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 384 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M50.8 452.1L19.2 477.4c-2.1 1.7-4.7 2.6-7.4 2.6C5.3 480 0 474.7 0 468.2V192C0 86 86 0 192 0S384 86 384 192V468.2c0 6.5-5.3 11.8-11.8 11.8c-2.7 0-5.3-.9-7.4-2.6l-31.6-25.3c-3.3-2.7-7.5-4.1-11.8-4.1c-5.9 0-11.5 2.8-15 7.5l-37.6 50.1c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4l-38.4-51.2c-3-4-7.8-6.4-12.8-6.4s-9.8 2.4-12.8 6.4l-38.4 51.2c-3 4-7.8 6.4-12.8 6.4s-9.8-2.4-12.8-6.4L77.6 455.5c-3.6-4.7-9.1-7.5-15-7.5c-4.3 0-8.4 1.5-11.7 4.1zM160 192c0-17.7-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32s32-14.3 32-32zm96 32c17.7 0 32-14.3 32-32s-14.3-32-32-32s-32 14.3-32 32s14.3 32 32 32z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </div>
      <div
        class="column"
      >
        <li
          class="faicon-link-container"
        >
          <a
            aria-label="social media link"
            class="faicon-link no-label is-black"
            data-cy="faicon-link"
            href="/"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-envelope fa-xl icon"
              data-icon="envelope"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M48 64C21.5 64 0 85.5 0 112c0 15.1 7.1 29.3 19.2 38.4L236.8 313.6c11.4 8.5 27 8.5 38.4 0L492.8 150.4c12.1-9.1 19.2-23.3 19.2-38.4c0-26.5-21.5-48-48-48H48zM0 176V384c0 35.3 28.7 64 64 64H448c35.3 0 64-28.7 64-64V176L294.4 339.2c-22.8 17.1-54 17.1-76.8 0L0 176z"
                fill="currentColor"
              />
            </svg>
          </a>
        </li>
      </div>
    </ul>
  </div>
</DocumentFragment>
`;
